### PR TITLE
test(evm): more e2e test contracts for edge cases

### PR DESCRIPTION
**Pull request by onikonychev**
Introduces 2 more test types:

1. Sending native (non ERC-20) tokens via smart contract
2. Executing smart contract with infinite loop to fail with "out of gas"

### CODE REVIEW

File: e2e/evm/README.md
Comment (Unique-Divine) [inline e2e/evm/README.md:1]: JavaScript is insanely slow

File: e2e/evm/test/basic_queries.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            value: amountToSend
+        }
+        const txResponse = await account.sendTransaction(transaction)
+        await txResponse.wait()
+        expect(txResponse).toHaveProperty('blockHash')
+
+        const senderBalanceAfter = await provider.getBalance(account.address)
+        const recipientBalanceAfter = await provider.getBalance(randomAddress)
+
+        // TODO: gas is not deducted regardless the gas limit, check this
```
Comment (Unique-Divine) [inline e2e/evm/test/basic_queries.test.js:30]:
```suggestion
        // TODO: https://github.com/NibiruChain/nibiru/issues/1902
        // gas is not deducted regardless the gas limit, check this
```

**Review by Unique-Divine — APPROVED** [review]
